src/components/Table/GridTable.tsx

```diff
@@ -1180,8 +1181,23 @@ function GridRow<R extends Kinded, S>(props: GridRowProps<R, S>): ReactElement {
   return openCardStyles && openCardStyles.length > 0 ? wrapCard(openCardStyles, rowNode) : rowNode;
 }
 
-// Fix to work with generics, see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37087#issuecomment-656596623
-const MemoizedGridRow = React.memo(GridRow) as typeof GridRow;
+/**
+ * Memoizes GridRows so that re-rendering the table doesn't re-render every single row.
+ *
+ * We use a custom `propsAreEqual` for the `GridRowProps.row` property, which we memoize
+ * based on the `GridDataRow.data` prop, such that if a table re-renders (i.e. for a cache
+ * updated) and technically creates new row instances, but a row's `row.data` points to the
+ * same/unchanged Apollo cache fragment, then we won't re-render that row.
+ *
+ * Note that if you're using virtualization, it can be surprising how unnoticeable broken row
+ * memoization is.
+ */
+// Declared as a const + `as typeof GridRow` to work with generics, see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37087#issuecomment-656596623
+const MemoizedGridRow = React.memo(GridRow, (one, two) => {
+  const { row: row1, ...others1 } = one;
+  const { row: row2, ...others2 } = two;
```

Comment:
**question:** Where is it that we are memoizing on `data`? From what I can tell here, we are still memoizing on the whole `GridDataRow`, not just `GridDataRow.data` when we destructure `row1` and `row2` out.

Reply:
`shallowEqual` compares keys+values within an object (kinda like props).

So we're treating `row1` (old) and `row2` (new) as "mini-props", and so `shallowEqual` effectively checks, for each key/value in `row1`/`row2`, are they the same? So is `id` the same, `data` the same, etc.

So even w/o an explicit `data1 === data2`, `shallowEqual` is effectively doing that for us, plus checking any other keys within each `GridDataRow` just in case something else changed.

I.e. you're right that we "memoize on the whole `GridDataRow`"--except now that we're memoizing on the key/values _within_ the row, instead of the row's hash/identity itself (which is what happens by default when it was just a `GridRow` prop with all the other `GridRow` props).

Reply:
Makes sense. Just the title of the PR threw me off, I guess. But I get what we're doing now. Thanks!

Reply:
Yep! Fwiw I had originally written it to `shallowEquals(row1.data, row2.data)` b/c that does seem the most intuitive, but rewrote it to `shallowEquals(row1, row2)` when I realized that'd work too, and also sanity check that some other row key (...pinned? something like that...) changes on us.